### ISSUE TITLE
Better packaging for the Python package

### ISSUE THREAD

**Issue by paugier** [turbulencelyon/4d-ptv#6]
I noted that the C++ code was twice in the repository (in Matching and in PSMN/Matching). 

I'd like to avoid this for the Python code. One should be able to install the package so that a command to run the script would be available from any directory.

I don't think it is necessary to upload a package on PyPI but it would be convenient to install (and build) everything with one standard command from the base directory of the project (typically `pip install -e .`).

What could be a good name for the Python package (which will potentially contain more than just the code for matching) and for the command to launch the matching code? To be honest, I'm not a fan of "STM".

Note: the name of a Python package cannot start by a figure so it cannot be 4dptv.

By the way, there are already some Python projects to do particle tracking:

- http://soft-matter.github.io/trackpy/v0.4.2/  and https://pypi.org/project/ptvpy/
- http://github.com/openptv/openptv-python
- https://perso.univ-rennes1.fr/joris.heyman/trac.html

Do you know the differences between these projects and what is done in 4d-ptv ?

**Comment by jsalort** [comment]
The details of the STM method and how it differs from other algorithms is described in a paper by Mickael Bourgoin and Sander Guisman, [Rev. Sci. Instrum. 91, 085105 (2020)](https://doi.org/10.1063/5.0009357).

I have added a short paragraph in the documentation, as well as a link to the published paper. Commit https://github.com/turbulencelyon/4d-ptv/commit/28a079a2fd8351adfba8d7e7e0ea602aea7e01b7
Probably we should have a general introduction in the documentation, with links to other similar projects, and possibly a broader list of differences. However, I don't know the other packages sufficiently well at the moment.

The general idea (in French), from discussion with Mickael:

> En gros la plupart des autres algo sont basés sur des projection de lignes epipolaires entre caméra avec souvent des choix du genre, on commence par une camera puis on projette sur les autres, et à la fin il faut faire des tests d’autoconsistance entre les différentes permutations. Ca rend les algos souvent lourd, dépendants de l’ordre des opérations et avec des scalings pas tres avantageux en fonction du nbre de camera. 
> L’algo STM s’affranchit un peu de tout ça et permet de matcher toutes les cameras en meme temps sans ordre prescrit et in fine avec des scalings pas trop mauvais.

As for the name, we are open to suggestions. In the long run, the package could also be uploaded on PyPI.
I don't know why the C++ ended up being there twice. This is probably a mistake. @ddumont-2607 ?

**Comment by ddumont-2607** [comment]
> I don't know why the C++ ended up being there twice. This is probably a mistake. @ddumont-2607 ?
Yes this is a mistake. C++ code is not necessary in the PSMN folder.

**Comment by jsalort** [comment]
> > I don't know why the C++ ended up being there twice. This is probably a mistake. @ddumont-2607 ?

> Yes this is a mistake. C++ code is not necessary in the PSMN folder.

OK, so, I can remove the `PSMN/Matching/STMCpp` folder entirely ?

**Comment by ddumont-2607** [comment]
> OK, so, I can remove the `PSMN/Matching/STMCpp` folder entirely ?

Yes.

**Comment by jsalort** [comment]
> > OK, so, I can remove the `PSMN/Matching/STMCpp` folder entirely ?
> 
> Yes.

Done in commit https://github.com/turbulencelyon/4d-ptv/commit/5c67865d83205ac858722ac152e3e6a49d957429
